01_fib/joel00wood/main_test.go

```diff
@@ -0,0 +1,147 @@
+		out <- buf.String()
+	}()
+	wg.Wait()
+	f()
+	writer.Close()
+	return <- out
+}
+// end borrowed code
+
+// isWholeNumber(n float64) wraps around math.Trunc and should return true for non-fractions, ergo whole numbers
+func TestIsWholeNumber(t *testing.T) {
+	testCases := map[string]struct {
+		input float64
+		expected bool
+	}{
+		"input 0": {0, true},
+		"input 1": {1, true},
+		"input 1.5": {1.5, false},
+		"input 0.5": {0.5, false},
+		"input 1000000000": {1000000000, true},
+	}
+
+	for name, test := range testCases {
+		actual := isWholeNumber(test.input)
+		if test.expected != actual {
+			t.Errorf("Unexpected response for %s iwWholeNumber(%f)\nExpected: %t\nActual: %t", name, test.input, test.expected, actual)
+		}
+	}
+}
+
+
+// fibCalc(n float64) calculates Nth fibonacci number, sign irrespective
+func TestFibCalc(t *testing.T) {
+	testCases := map[string]struct {
+		input, expected float64
+	}{
+		"input 20": {20, 6765},
+		"input 3": {3, 2},
+		"input 2": {2, 1},
+		"input 1": {1, 1},
+		"input 0": {0, 0},
+	}
+
+	for name, test := range testCases {
+		actual := fibCalc(test.input)
+		if test.expected != actual {
+			t.Errorf("Unexpected output for %s fibCalc(%f)\nExpected: %f\nActual: %f", name, test.input, test.expected, actual)
+		}
+	}
+}
+
+// fibN(n float64) calculates Nth fibonacci number, sign respective (ergo supports negafibonacci
+func TestFibN(t *testing.T) {
+	testCases := map[string]struct {
+		input, expected float64
+	}{
+		"input 20": {20, 6765},
+		"input 3": {3, 2},
+		"input 2": {2, 1},
+		"input 1": {1, 1},
+		"input 0": {0, 0},
+		"input -1": {-1, 1},
+		"input -2": {-2, -1},
+		"input -3": {-3, 2},
+		"input -20": {-20, -6765},
+	}
+
+	for name, test := range testCases {
+		actual := fibN(test.input)
+		if test.expected != actual {
+			t.Errorf("Unexpected output for %s fibN(%f)\nExpected: %f\nActual: %f", name, test.input, test.expected, actual)
+		}
+	}
+}
+
+// fib(n float64) should output range from 0 to Nth fibonacci number, one per line, ending in a new line
+func TestFib(t *testing.T) {
+	testCases := map[string]struct {
+		input float64
+		expected string
+	}{
+		"input 1.5": {1.5, "Invalid input, expected whole number: 1.5\n"},
+		"input 7": {7, "0\n1\n1\n2\n3\n5\n8\n13\n"},
+		"input 3": {3, "0\n1\n1\n2\n"},
+		"input 2": {2, "0\n1\n1\n"},
+		"input 1": {1, "0\n1\n"},
+		"input 0": {0, "0\n"},
+		"input -1": {-1, "0\n1\n"},
+		"input -2": {-2, "0\n1\n-1\n"},
+		"input -3": {-3, "0\n1\n-1\n2\n"},
+		"input -7": {-7, "0\n1\n-1\n2\n-3\n5\n-8\n13\n"},
+		"input -1.5": {-1.5, "Invalid input, expected whole number: -1.5\n"},
+	}
+	for name, test := range testCases {
+		actual := captureOutput(func() {fib(test.input)})
+		if test.expected != actual {
+			t.Errorf("Unexpected response for %q fib(%f)\nExpected: %q\nActual: %q", name, test.input, test.expected, actual)
+		}
+	}
+}
+
+
+// test main() which is essentially fib(7)
+func TestMain(t *testing.T) {
+	expected := "0\n1\n1\n2\n3\n5\n8\n13\n"
```

Comment:
You can use:
```
expected := `0
1
1
.
`
```
And then filter out the \n with strings.Replace in the actual:
`actual := captureOutput(strings.Replace(func() {main()}, "\n", " ", -1))`

Reply:
In this case I'm happy to use the `\n` characters as it reads better and is easier to manage in the test framework